portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -618,5 +647,181 @@ public String getSkinPrefix() {
 		return "";
 	}
 
+	public List<BullhornAlert> getSocialAlerts(String userId) {
+
+		List<BullhornAlert> alerts = sqlService.dbRead(
+                "SELECT * FROM SOCIAL_ALERTS WHERE TO_USER = ? ORDER BY EVENT_DATE DESC"
```

Comment:
I'm not suggesting it would be a performance issue, rather a code maintenance issue. Say you were to use this query in two places, you would then hard-code it in both places. Then you discover a bug, and the query needs tweaking, so you update the SQL in one place but not the other.

It just creates the opportunity for human error in my opinion. However, I'd be curious to hear how others feel about this.